include/zephyr/sys_clock.h

```diff
@@ -79,6 +79,36 @@ typedef struct {
  */
 #define K_TIMEOUT_EQ(a, b) ((a).ticks == (b).ticks)
 
+/** number of nsec per usec */
+#define NSEC_PER_USEC 1000U
+
+/** number of nsec per msec */
+#define NSEC_PER_MSEC 1000000U
+
+/** number of microseconds per millisecond */
+#define USEC_PER_MSEC 1000U
+
+/** number of milliseconds per second */
+#define MSEC_PER_SEC 1000U
+
+/** number of seconds per minute */
+#define SEC_PER_MIN 60U
+
+/** number of minutes per hour */
+#define MIN_PER_HOUR 60U
+
+/** number of hours per day */
+#define HOUR_PER_DAY 24U
+
+/** number of microseconds per second */
+#define USEC_PER_SEC ((USEC_PER_MSEC) * (MSEC_PER_SEC))
+
+/** number of nanoseconds per second */
+#define NSEC_PER_SEC ((NSEC_PER_USEC) * (USEC_PER_MSEC) * (MSEC_PER_SEC))
```

Comment:
maybe for a later update but these time-conversion related constants might be worth grouping in a `@name`d section. There are in fact probably several groups of such macros that could be grouped -- see below.
I know that you want to work on this incrementally but should you catch low-hanging fruits with regards to adding more structure where you can, please go for it :)

<img width="398" alt="image" src="https://github.com/zephyrproject-rtos/zephyr/assets/128251/90f6495d-32b1-411a-a7dd-acca2c78c785">

Reply:
yeah, will do in a second round, right now I want to focus on removing private/internal mostly and low hanging fruit, ie.. misconfigured doxygen comments, not add comments myself and go into loops of reviews about documentation and content that I might get wrong.